api/clients/v2/verification/cert_verifier.go

```diff
@@ -15,58 +16,29 @@ import (
 	gethcommon "github.com/ethereum/go-ethereum/common"
 )
 
-// ICertVerifier is the interface representing a CertVerifier
-//
-// This interface exists in order to allow verification mocking in unit tests.
-type ICertVerifier interface {
-	// VerifyCertV2 calls the VerifyCertV2 view function on the EigenDACertVerifier contract.
-	//
-	// This method returns nil if the cert is successfully verified. Otherwise, it returns an error.
-	VerifyCertV2(
-		ctx context.Context,
-		certVerifierAddress string,
-		eigenDACert *EigenDACert,
-	) error
-
-	// GetNonSignerStakesAndSignature calls the getNonSignerStakesAndSignature view function on the EigenDACertVerifier
-	// contract, and returns the resulting NonSignerStakesAndSignature object.
-	GetNonSignerStakesAndSignature(
-		ctx context.Context,
-		certVerifierAddress string,
-		signedBatch *disperser.SignedBatch,
-	) (*verifierBindings.NonSignerStakesAndSignature, error)
-
-	// GetQuorumNumbersRequired queries the cert verifier contract for the configured set of quorum numbers that must
-	// be set in the BlobHeader, and verified in VerifyDACertV2 and verifyDACertV2FromSignedBatch
-	GetQuorumNumbersRequired(
-		ctx context.Context,
-		certVerifierAddress string,
-	) ([]uint8, error)
-}
-
 // CertVerifier is responsible for making eth calls against the CertVerifier contract to ensure cryptographic and
 // structural integrity of V2 certificates
 //
 // The cert verifier contract is located at https://github.com/Layr-Labs/eigenda/blob/master/contracts/src/core/EigenDACertVerifier.sol
 type CertVerifier struct {
-	logger       logging.Logger
-	ethClient    common.EthClient
-	pollInterval time.Duration
-	// storage shared between goroutines, containing the most recent block number observed by calling ethClient.BlockNumber()
-	latestBlockNumber atomic.Uint64
-	// atomic bool, so that only a single goroutine is polling the internal client with BlockNumber() calls at any given time
-	pollingActive atomic.Bool
+	logger                      logging.Logger
+	ethClient                   common.EthClient
+	blockNumberProvider         *BlockNumberProvider
+	certVerifierAddressProvider clients.CertVerifierAddressProvider
 	// maps contract address to a ContractEigenDACertVerifierCaller object
 	verifierCallers sync.Map
+	// maps cert verifier address string to set of required quorums specified in the contract at that address
```

Comment:
nit: use same doc for requiredQuorums and verifierCallers

`maps contract address`
vs
`maps cert verifier address`

to make it obvious they are the same. consider also potentially merging into a single map whose value is a struct containing both caller + requiredQuorums?

Reply:
Modified comment in [947a1b7e](https://github.com/Layr-Labs/eigenda/pull/1368/commits/947a1b7e45155cf039725d1b8a5c2bb1c52112a8)

I think it would increase the complexity of the logic to combine these maps into one: depending on the needs of the user, it's possible that they might end up populating the verifierCallers, but not the required quorums (if they are verifying, but not dispersing blobs). By keeping these maps separate, it's possible to infer from map membership whether the needed value exists, without needing to check the map value.